core/src/types/list.rs

```diff
@@ -80,43 +83,87 @@ impl Stream for Lister {
             return Poll::Ready(None);
         }
 
-        if let Some(fut) = self.stating.as_mut() {
-            let (path, rp) = ready!(fut.poll_unpin(cx));
-
-            // Make sure we will not poll this future again.
-            self.stating = None;
-            let metadata = match rp {
-                Ok(rp) => rp.into_metadata(),
-                Err(err) => {
-                    self.errored = true;
-                    return Poll::Ready(Some(Err(err)));
-                }
-            };
-
-            return Poll::Ready(Some(Ok(Entry::new(path, metadata))));
+        if !self.task_queue.is_empty() {
+            if let Some(handle) = self.task_queue.back_mut() {
+                let (path, rp) = ready!(handle.poll_unpin(cx)).map_err(|err| {
+                    Error::new(ErrorKind::Unexpected, "join handle error")
```

Comment:
We can use `opendal::raw::new_task_join_error`

Reply:
Done###